drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_data_element_helper.rb

```diff
@@ -51,26 +51,7 @@ def new_cde_record(value:, owner_type:, owner_id:, definition_key:, date_created
         DateUpdated: date_created,
         data_source_id: data_source.id,
         UserID: system_user.id,
-      }.merge(cde_value_fields(definition, value))
-    end
-
-    VALUE_FIELDS = [
-      'float',
-      'integer',
-      'boolean',
-      'string',
-      'text',
-      'date',
-      'json',
-    ].map { |v| [v, "value_#{v}"] }
-
-    # note, we need to set all fields- bulk insert becomes unhappy if the columns are not uniform
-    def cde_value_fields(definition, value)
-      result = {}
-      VALUE_FIELDS.map do |field_type, field_name|
-        result[field_name] = field_type == definition.field_type ? value : nil
-      end
-      result
+      }.merge(HmisExternalApis::ExternalForms::FormSubmission.cde_value_fields(definition, value))
```

Comment:
@ttoomey I know this is code used by the loader and we should avoid changing it, but I think your suggestion in a previous comment was to extract this into the FormSubmission. Let me know if you think the change I have made here looks ok or should be reverted (resulting in duplicated code, but will be removed when we remove all the loader code).